source/graphql/helper.d

```diff
@@ -722,31 +722,23 @@ template isNotInTypeSet(T, R...) {
 }
 
 string getTypename(Schema,T)(auto ref T input) @trusted {
```

Comment:
The original function was iterating over all types in the `collectTypes` list, and checking if the runtime type of input could be converted to the type. In this case, I'm using the already-existing `TypeInfo` linked list to find any types that it supports.

But I see another issue in my implementation -- we should support interfaces as well. I need to check those typeinfos.

I'll not also that there is an issue with the way it was done. In the original, once ANY type that is convertible is found, it returns. But there may be a more suitable choice. For example if you have:

```D
class A {}
class B : A {}
class C : B {}
```

The function which receives an instance of C, typed as A, it might return "B" and not "C", even if `C` is in the list.

Reply:
sounds good

Reply:
OK, actually, I think this is fine the way it is. The original purposely did not search for implementing interfaces, as it removed everything but classes from the collected types.